### ISSUE TITLE
CORS error using web service

### ISSUE THREAD

**Issue by smalers** [OpenWaterFoundation/owf-app-infomapper-ng#294]
Trying to use a REST web service that returns GeoJSON resulted in the following error.  The URL is:

[http://bridge-scour.trilynx-novastar.systems/novastar/data/api/v1/stations?format=geojson](http://bridge-scour.trilynx-novastar.systems/novastar/data/api/v1/stations?format=geojson)

The web service sets `Access-Control-Allow-Origin=*` and web services are used successfully in other Angular applications.  Does InfoMapper need to have something changed?   It seems like the above should fix it.  Note also that the legend does not show the error indicator for the bad layer.  Need to troubleshoot this issue.

![image](https://user-images.githubusercontent.com/6448274/94352572-fccbc480-0023-11eb-856a-717eb7bec204.png)

**Comment by smalers** [comment]
I tested using Postman and the `Access-Control-Allow-Origin=*` header property is set.  I'm going to reach out to Nathan and see if the server has been configured somehow, although it is not obvious to me why behavior in an app would be different than the browser.

**Comment by Nightsphere** [comment]
After some research, I feel like I've gotten closer to the issue, but I'm not quite there yet. According to mozilla's website, the `*` value for the `Access-Control-Allow-Methods` only works if a certain attribute name `crossorigin` is set in the request. I've looked into trying to send that in the GET request, but haven't found a way yet. Another interesting thing I've discovered is that there might be some headers that also need to be sent in the request, such as `Access-Control-Request-Method: GET`.

Oddly, the error presented in the console was that the response header did not contain `Allow-Control-Allow-Origin`, and indeed, the headers from the server were as follows:

![image](https://user-images.githubusercontent.com/22484403/95513502-b194a880-0977-11eb-96e6-d76377a18db8.png)

This might be because of the lack of request headers discussed above, however. Lastly, I changed the http beginning of the URL with https and it seemed to work, but I'm not sure if it's a sustainable solution. I think more research and testing will have to be done.
